java/java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -1395,15 +1393,12 @@ private static File dumpHeap(@NonNull final String path) {
     private static Pair<Object,Method> heapDumper;
 
     private static String computeJavacVersion() {
-        if (NoJavacHelper.hasNbJavac()) {
-            File nbJavac = InstalledFileLocator.getDefault().locate("modules/ext/nb-javac-impl.jar", "org.netbeans.modules.nbjavac.impl", false);
-            if (nbJavac != null) {
-                return String.valueOf(nbJavac.lastModified());
+        for (ModuleInfo mi : Lookup.getDefault().lookupAll(ModuleInfo.class)) {
+            if ("org.netbeans.libs.nbjavacapi".equals(mi.getCodeNameBase())) {
+                return mi.getSpecificationVersion().toString();
             }
-            return "-1";
-        } else {
-            return System.getProperty("java.vm.version", "unknown");
         }
+        return System.getProperty("java.version", "unknown");
     }
```

Comment:
please note that the nb-javac module spec version is `feature.interim.update` (17.0.1), which is the same as the `java.version` property.

This would have the effect of triggering a re-index if a nb-javac update is delivered via the plugin portal.

a potential alternative would be simply returning `SourceVersion.latest()` which is feature version only (would also handle nb-javac and javac the same).

Reply:
the module lookup code caused module loading failures in the tests. I am going to return `SourceVersion` temporarily and see if this alone is enough to fix travis (it works locally).

Reply:
tests are green without the module lookup.